sphinxcontrib/httpdomain.py

```diff
@@ -692,6 +695,27 @@ def get_objects(self):
                 anchor = http_resource_anchor(method, path)
                 yield (path, path, method, info[0], anchor, 1)
 
+    def merge_domaindata(self, docnames, otherdata):
+        """Merge domaindata from the workers/chunks when they return.
+
+        Called once per parallelization chunk.
+        Only used when sphinx is run in parallel mode.
+
+        :param docnames: a Set of the docnames that are part of the current chunk to merge
+        :param otherdata: the partial data calculated by the current chunk
+        """
+        for typ in self.object_types:
+            self_data = self.data[typ]
+            other_data = otherdata[typ]
+            for entry_point_name, entry_point_data in other_data.items():
```

Comment:
My testing reveals that if you use `.. only:: <tag>` to conditionally include documents in contents, that for some reason, Sphinx flags those as being duplicate definitions. A simple example of my toc in the base document:
```
.. toctree::
   :hidden:

   about

.. only:: internal

   .. toctree::
      :hidden:

      internal/apiprop

.. toctree::
   :hidden:

   src/tools
```

I really don't know why this is happening, but I suspect it's yet another example of how the `.. only::` directive isn't _really_ a conditional include fully because of Sphinx's multipass processing. 8(

I'm not sure how widely used this trick is of conditionally including documents in a build...

Reply:
Yes, I think this is a sphinx problem that is already discussed at some places. A simple
```
.. _xyz:

.. only:: bla

   .. _xyz:
```
already exposes the same behaviour so I guess this has nothing to do with our changes .. ?

Reply:
It's possible; what's weird to me is that the warnings I got from your code about HTTP URL/routes defined more than once are not correct -- they are defined only once in my source, but all the definitions included in the TOC via that `.. only::` are somehow getting noted as "duplicates".

I agree that this is probably more to do with the way that `.. only::` works and less to do with your implementation -- it's just that your implementation notices this duplication and issues a warning for it... I bet the builder's already doing this and we just overwrite that "duplication" when merging the domain data silently. 8/

Reply:
New maintainer here :wave:

Looking at the sphinx docs for [:only;](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html?highlight=only#directive-only) it says

> *Warning*
> This directive is designed to control only content of document. It could not control sections, labels and so on.

I think this means your use of only is unsupported @ViktorHaag.

Reply:
I can believe it's unsupported; having conditional inclusion in a `toc` is very useful, though... 8/

Reply:
Perhaps we could have a config setting for this extension to not warn about duplicates?

Reply:
@ViktorHaag If you want this merge behaviour, please open a new PR to add it (back) -- I'm  merging this PR now.